app/src/main/java/org/systers/mentorship/view/activities/LoginActivity.kt

```diff
@@ -2,131 +2,234 @@ package org.systers.mentorship.view.activities
+    val passwordFocusRequest = remember { FocusRequester() }
+
+    Box {
+        Column(
+            Modifier
+                .fillMaxSize()
+                .verticalScroll(rememberScrollState()),
+            verticalArrangement = Arrangement.SpaceAround,
+            horizontalAlignment = Alignment.CenterHorizontally
+        ) {
+
+            // This LaunchedEffect will display a snackbar each time the snackbar message changes
+            LaunchedEffect(key1 = snackbarMessage) {
+                scope.launch {
+                    snackbarMessage?.let {
+                        if (snackbarMessage.isNotEmpty()) {
+                            scaffoldState.snackbarHostState.showSnackbar(
+                                message = it,
+                                duration = SnackbarDuration.Short
+                            )
+                        }
+                    }
+                }
+            }
 
-    private fun login() {
-        CountingIdlingResourceSingleton.increment()
-        username = tiUsername.editText?.text.toString().trim()
-        password = tiPassword.editText?.text.toString().trim()
-        if (validateCredentials()) {
-            loginViewModel.login(Login(username, password))
-            showProgressDialog(getString(R.string.logging_in))
-        } else {
-            CountingIdlingResourceSingleton.decrement()
+            // The mentorship system logo
+            Image(
+                painter = painterResource(id = R.drawable.mentorship_system_logo),
+                contentDescription = "Mentorship System Logo",
+                modifier = Modifier.fillMaxWidth(0.95f)
+            )
+
+            Column {
+
+                EmailTextField(
+                    value = username,
+                    onUsernameChange = onUsernameChange,
+                    enabled = username.isEmpty() || password.isEmpty() || isButtonEnabled,
+                    onImeAction = { passwordFocusRequest.requestFocus() },
+                    modifier = Modifier.focusRequester(signUpFocusRequest)
+                )
+
+                PasswordTextField(
+                    value = password,
+                    modifier = Modifier
+                        .focusRequester(passwordFocusRequest),
+                    onPasswordChange = onPasswordChange,
+                    enabled = username.isEmpty() || password.isEmpty() || isButtonEnabled,
+                    onImeAction = { onLoginButtonClick(); focusManager.clearFocus() }
+                )
+            }
+            Column {
+                Button(
+                    onClick = onLoginButtonClick,
+                    enabled = isButtonEnabled,
+                    shape = MaterialTheme.shapes.medium,
+                    modifier = Modifier.fillMaxWidth(0.5f)
+                ) {
+                    Text(text = stringResource(R.string.login))
+                }
+                Spacer(modifier = Modifier.height(10.dp))
```

Comment:
we should keep those values in themes I think, not hardcoded in views.

Reply:
Sure I'll refractor this